### ISSUE TITLE
Expose Kubernetes api services via F5 ingress crashes CIS

### ISSUE THREAD

**Issue by f5minions** [F5Networks/k8s-bigip-ctlr#1391]
#### Setup Details
 CIS Version : 2.0      
 Build: f5networks/k8s-bigip-ctlr
 BIGIP Version: Big IP 14.1.2      
 AS3 Version: 3.20     
 Agent Mode: AS3       
 Orchestration: K8S     
 Orchestration Version: 1.17 and 1.18    
 Pool Mode: Cluster      
 Additional Setup details: <Platform/CNI Plugins/ cluster nodes/ etc>

#### Description  
when i configured ingress for kubernetes API server than CIS controller was stop working and become crashloopback status. I used below ingres YAML,

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    virtual-server.f5.com/health: |
      [
        {
          "path":     "kube-apiserver.skb-doj-dev01.btvpaas.com/",
          "send":     "HTTP GET /",
          "interval": 5,
          "timeout":  10
        }
      ]
    virtual-server.f5.com/ip: 175.113.150.30
    virtual-server.f5.com/partition: k8s-dev
  name: kube-apiserver
  namespace: default
spec:
  rules:
  - host: kube-apiserver.skb-doj-dev01.btvpaas.com
    http:
      paths:
      - backend:
          serviceName: kubernetes
          servicePort: 443
        path: /
```
 

And CIS controller logs is below,
Controller log-----------------------------------------------
 
[centos@doj-dw03 ~]$ sudo docker logs f1cff628b6f5
2020/07/06 12:28:27 [INFO] [INIT] Starting: Container Ingress Services - Version: 2.0.0, BuildInfo: cloud-user-efcab9e-20200522033624
2020/07/06 12:28:27 [INFO] ConfigWriter started: 0xc00030c480
2020/07/06 12:28:27 [INFO] Started config driver sub-process at pid: 16
2020/07/06 12:28:27 [INFO] [INIT] Creating Agent for as3
2020/07/06 12:28:27 [INFO] [AS3] Initializing AS3 Agent
2020/07/06 12:28:27 [INFO] [CCCL] Initializing CCCL Agent
2020/07/06 12:28:27 [INFO] [CORE] NodePoller (0xc00011eab0) registering new listener: 0x130d7c0
2020/07/06 12:28:27 [INFO] [CORE] NodePoller started: (0xc00011eab0)
2020/07/06 12:28:27 [INFO] [CORE] Watching Ingress resources.
2020/07/06 12:28:27 [INFO] [CORE] Watching ConfigMap resources.
2020/07/06 12:28:27 [INFO] [CORE] Handling ConfigMap resource events.
2020/07/06 12:28:27 [INFO] [CORE] Handling Ingress resource events.
2020/07/06 12:28:27 [INFO] [CORE] Registered BigIP Metrics
E0706 12:28:27.746474       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 165 [running]:
github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x143a760, 0x25b7e30)
        /build/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /build/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x143a760, 0x25b7e30)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).getEndpointsForCluster(0xc0003e6000, 0xc00015fb40, 0x5, 0xc000613db0, 0xc000613db0, 0xc00015ed01, 0x0)
        /build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:2230 +0x39d
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).updatePoolMembersForCluster(0xc0003e6000, 0xc0001ac5e8, 0xc00015ed80, 0xa, 0x1bb, 0xc00015fa88, 0x7, 0xc0007329c0, 0xc0004a3880, 0x0, ...)
        /build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1900 +0x1fc
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).handleConfigForType(0xc0003e6000, 0xc0007329c0, 0xc00015fa88, 0x7, 0xc00015fa90, 0xa, 0x0, 0x0, 0x0, 0x0, ...)
        /build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1822 +0xa4c
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).syncIngresses(0xc0003e6000, 0xc000707a40, 0xc00015fa88, 0x7, 0xc00015fa90, 0xa, 0x0, 0x0, 0x0, 0x0, ...)
        /build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1320 +0xde6
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).syncVirtualServer(0xc0003e6000, 0xc00015fa88, 0x7, 0xc00015fa90, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:964 +0x9f5
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).processNextVirtualServer(0xc0003e6000, 0xc00052db00)
        /build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:879 +0x15b
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).virtualServerWorker(0xc0003e6000)
        /build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:825 +0x2b
github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000499cb0)
        /build/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x54
github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000499cb0, 0x3b9aca00, 0x0, 0x1, 0xc00054c240)
        /build/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc000499cb0, 0x3b9aca00, 0xc00054c240)
        /build/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d

So if some customer want to use F5 ingress for their kubernetes API server what's recommended way to achieve it? 

#### Steps To Reproduce
1) If you using above ingress YAML, it always will happen.
2)
3)

#### Expected Result 
Want to know we can use ingress for kubernetes api server or we can just use manual configuration for that services? 

#### Actual Result  
please check what's cause of this issue.

#### Diagnostic Information
```buildoutcfg
<Configuration files, error messages, logs>
Note: Sanitize the data. For example, be mindful of IPs, ports, application names and URLs
Note: The following F5 article outlines the information required when opening an issue.
https://support.f5.com/csp/article/K60974137
```

**Comment by cisbotctlr** [comment]
cisbot will assign the issue to one of the devs. 
 @devs, use /jira for internal tracking.

**Comment by mdditt2000** [comment]
afternoon @f5minions please can you modify your Ingress API to the new ingress API

apiVersion: networking.k8s.io/v1beta1
kind: Ingress

Here is a working example

https://github.com/mdditt2000/kubernetes-1-18/blob/master/cis%202.0/ingress/big-ip-92-cluster/ingress-examples/f5-k8s-ingress-http.yaml

Let me know if that resolves the problem. Thanks Mark

**Comment by f5minions** [comment]
Hi mdditt,
i tried to new ingress api but result is same like below,

E0708 14:22:48.755966       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 104 [running]:
github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x143a760, 0x25b7e30)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x143a760, 0x25b7e30)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).getEndpointsForCluster(0xc0003a6000, 0xc0003d3c65, 0x5, 0xc0001bb690, 0xc0001bb690, 0xc000863b01, 0x0)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:2230 +0x39d
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).updatePoolMembersForCluster(0xc0003a6000, 0xc0006bcdc8, 0xc000863bb0, 0xa, 0x1bb, 0xc000863ab0, 0x7, 0xc0000d01a0, 0xc0003bc400, 0x0, ...)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1900 +0x1fc
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).handleConfigForType(0xc0003a6000, 0xc0000d01a0, 0xc000863ab0, 0x7, 0xc000863bb0, 0xa, 0x0, 0x0, 0x0, 0x0, ...)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1822 +0xa4c
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).syncIngresses(0xc0003a6000, 0xc0004c3a40, 0xc000863ab0, 0x7, 0xc000863bb0, 0xa, 0x0, 0x0, 0x0, 0x0, ...)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1320 +0xde6
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).syncVirtualServer(0xc0003a6000, 0xc000863ab0, 0x7, 0xc000863bb0, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:964 +0x9f5
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).processNextVirtualServer(0xc0003a6000, 0xc0002c4700)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:879 +0x15b
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).virtualServerWorker(0xc0003a6000)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:825 +0x2b
github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000186c90)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x54
github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000186c90, 0x3b9aca00, 0x0, 0x1, 0xc000349e00)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc000186c90, 0x3b9aca00, 0xc000349e00)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).runImpl
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:769 +0x13d
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12abb5d]

goroutine 104 [running]:
github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x105
panic(0x143a760, 0x25b7e30)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).getEndpointsForCluster(0xc0003a6000, 0xc0003d3c65, 0x5, 0xc0001bb690, 0xc0001bb690, 0xc000863b01, 0x0)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:2230 +0x39d
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).updatePoolMembersForCluster(0xc0003a6000, 0xc0006bcdc8, 0xc000863bb0, 0xa, 0x1bb, 0xc000863ab0, 0x7, 0xc0000d01a0, 0xc0003bc400, 0x0, ...)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1900 +0x1fc
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).handleConfigForType(0xc0003a6000, 0xc0000d01a0, 0xc000863ab0, 0x7, 0xc000863bb0, 0xa, 0x0, 0x0, 0x0, 0x0, ...)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1822 +0xa4c
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).syncIngresses(0xc0003a6000, 0xc0004c3a40, 0xc000863ab0, 0x7, 0xc000863bb0, 0xa, 0x0, 0x0, 0x0, 0x0, ...)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1320 +0xde6
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).syncVirtualServer(0xc0003a6000, 0xc000863ab0, 0x7, 0xc000863bb0, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:964 +0x9f5
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).processNextVirtualServer(0xc0003a6000, 0xc0002c4700)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:879 +0x15b
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).virtualServerWorker(0xc0003a6000)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:825 +0x2b
github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000186c90)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x54
github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000186c90, 0x3b9aca00, 0x0, 0x1, 0xc000349e00)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc000186c90, 0x3b9aca00, 0xc000349e00)
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).runImpl
	/build/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:769 +0x13d

And CIS controller was crashed again,

NAME                                          READY   STATUS             RESTARTS   AGE
calico-kube-controllers-76d4774d89-9jbql      1/1     Running            0          16d
calico-node-k557d                             1/1     Running            0          16d
calico-node-k5bgm                             1/1     Running            0          16d
calico-node-t72s7                             1/1     Running            0          16d
coredns-66bff467f8-8x4dw                      1/1     Running            0          16d
coredns-66bff467f8-zwtmx                      1/1     Running            0          16d
etcd-k8s-node1                                1/1     Running            0          16d
k8s-bigip1-ctlr-deployment-85c8575d6d-wrzjm   0/1     CrashLoopBackOff   4          22h
kube-apiserver-k8s-node1                      1/1     Running            0          16d
kube-controller-manager-k8s-node1             1/1     Running            1          16d
kube-proxy-7sbs7                              1/1     Running            0          16d
kube-proxy-dqg92                              1/1     Running            0          16d
kube-proxy-xjrhf                              1/1     Running            0          16d
kube-scheduler-k8s-node1                      1/1     Running            2          16d

We can reproduce it with my YAML. I guess that we can't use ingress for kubernetes API server. please check this once agian.

Thanks.

**Comment by mdditt2000** [comment]
@veeramalla-f5 any idea why we are seeing CIS CrashLoopBack?

**Comment by mdditt2000** [comment]
Please correct the path "kube-apiserver.skb-doj-dev01.btvpaas.com/", 

Again remove the / from the path.

**Comment by mdditt2000** [comment]
@f5minions I did some more testing and fixed the ingress to work with CIS. Please clone the repo to use a template

https://github.com/mdditt2000/kubernetes-1-18/blob/master/cis%202.1/github/1391/ingress.yaml

Ok with that said their is going to be issues with your path. You will need to shorten the path/hostname. The path kube-apiserver.skb-doj-dev01.btvpaas.com is to long to specify in the monitor. AS3 wont allow it due to character length.

Doing a get on BIG-IP you can see this. 

Create the ingress and described the ingress. 

```
[root@k8s-1-18-master 1391]# kubectl create -f ingress.yaml
ingress.networking.k8s.io/f5-hello-world created
[root@k8s-1-18-master 1391]# kubectl describe ingress f5-hello-world
Name:             f5-hello-world
Namespace:        default
Address:          10.192.75.108
Default backend:  default-http-backend:80 (<error: endpoints "default-http-backend" not found>)
Rules:
  Host                                      Path  Backends
  ----                                      ----  --------
  kube-apiserver.skb-doj-dev01.btvpaas.com
                                            /mysite/app1   f5-hello-world:8080 (10.244.2.79:8080,10.244.2.80:8080)
Annotations:                                virtual-server.f5.com/health:
                                              [
                                                {
                                                  "path": "kube-apiserver.skb-doj-dev01.btvpaas.com/",
                                                  "send": "HTTP GET /",
                                                  "interval": 5,
                                                  "timeout": 10
                                                }
                                              ]
                                            virtual-server.f5.com/ip: 10.192.75.108
Events:
  Type    Reason               Age                 From            Message
  ----    ------               ----                ----            -------
  Normal  ResourceConfigured   43s (x2 over 43s)   k8s-bigip-ctlr  Created a ResourceConfig 'ingress_10-192-75-108_80' for the Ingress.
  Normal  MonitorRuleNotFound  36s (x12 over 43s)  k8s-bigip-ctlr  Rule not found for Health Monitor path 'kube-apiserver.skb-doj-dev01.btvpaas.com/'
  Normal  MonitorRuleNotUsed   36s (x11 over 43s)  k8s-bigip-ctlr  Health Monitor path '' does not match any Ingress paths.
[root@k8s-1-18-master 1391]#
```

You can see those errors regarding the path. Path is to long. Look at the AS3 declaration

```
{
    "k8s": {
        "Shared": {
            "Ingress_10_192_75_108_80": {
                "class": "Endpoint_Policy",
                "rules": [
                    {
                        "name": "ingress_kube_apiserver_skb_doj_dev01_btvpaas_com_mysite_app1_ingress_default_f5_hello_world",
                        "conditions": [
                            {
                                "type": "httpHeader",
                                "name": "host",
                                "event": "request",
                                "all": {
                                    "values": [
                                        "kube-apiserver.skb-doj-dev01.btvpaas.com"
                                    ],
                                    "operand": "equals"
                                }
                            },
                            {
                                "type": "httpUri",
                                "name": "1",
                                "event": "request",
                                "index": 1,
                                "pathSegment": {
                                    "values": [
                                        "mysite"
                                    ],
                                    "operand": "equals"
                                }
                            },
                            {
                                "type": "httpUri",
                                "name": "2",
                                "event": "request",
                                "index": 2,
                                "pathSegment": {
                                    "values": [
                                        "app1"
                                    ],
                                    "operand": "equals"
                                }
                            }
                        ],
                        "actions": [
                            {
                                "type": "forward",
                                "event": "request",
                                "select": {
                                    "pool": {
                                        "use": "ingress_default_f5_hello_world"
                                    }
                                }
                            }
                        ]
                    }
                ],
                "strategy": "first-match"
            },
            "class": "Application",
            "ingress_10_192_75_108_80": {
                "layer4": "tcp",
                "source": "0.0.0.0/0",
                "translateServerAddress": true,
                "translateServerPort": true,
                "class": "Service_HTTP",
                "virtualAddresses": [
                    "10.192.75.108"
                ],
                "virtualPort": 80,
                "snat": "auto",
                "policyEndpoint": "/k8s/Shared/Ingress_10_192_75_108_80"
            },
            "ingress_default_f5_hello_world": {
                "class": "Pool",
                "loadBalancingMode": "round-robin",
                "members": [
                    {
                        "addressDiscovery": "static",
                        "serverAddresses": [
                            "10.244.2.79"
                        ],
                        "servicePort": 8080
                    },
                    {
                        "addressDiscovery": "static",
                        "serverAddresses": [
                            "10.244.2.80"
                        ],
                        "servicePort": 8080
                    }
                ]
            },
            "template": "shared"
        },
        "class": "Tenant"
    },
        "class": "Tenant"
    },
    "class": "ADC",
    "controls": {
        "class": "Controls",
        "userAgent": "CIS/v2.1.0-WIP K8S/v1.18.0",
        "archiveTimestamp": "2020-07-09T20:47:03.894Z"
    },
    "id": "urn:uuid:85626792-9ee7-46bb-8fc8-4ba708cfdc1d",
    "label": "CIS Declaration",
    "remark": "Auto-generated by CIS",
    "schemaVersion": "3.20.0",
    "updateMode": "selective"
}
```
Note monitor doesn't get created. 

Action: please shorten the host and everything will work

**Comment by mdditt2000** [comment]
Here is the example using secret for edge and serverssl profile for re-encrypt for https://github.com/mdditt2000/kubernetes-1-18/blob/master/cis%202.1/slack/ingress/re-encrypt-tls.yaml AS3 is also their

**Comment by f5minions** [comment]
Hi mdditt,
Please check this issue using below kubernetes internal service, named "kubenetes". If we using the other services such as f5-hello-world, there are no issue with F5 CIS. 

f5demo@k8s-node1:~/skt$ kubectl get svc
NAME             TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
f5-hello-world   NodePort    10.110.3.36     <none>        8080:32707/TCP   19d
httpbin1         ClusterIP   10.105.255.23   <none>        8000/TCP         16d
kubernetes       ClusterIP   10.96.0.1       <none>        443/TCP          19d. --> please use this internal service for ingress.



spec:
  rules:
  - host: kube-apiserver.skb-doj-dev01.btvpaas.com
    http:
      paths:
      - backend:
          serviceName: kubernetes --> this is crash points
          servicePort: 443
        path: /


Anyway i will check monitor's name once again, but this crash issue happen by "kubernetes" service. so please use above service on your lab. you may happen same issue.

Thanks.

**Comment by mdditt2000** [comment]
user want to create a ingress for the **service kubernetes**

```
[root@k8s-1-18-master ~]# kubectl get service kubernetes
NAME         TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
kubernetes   ClusterIP   10.96.0.1    <none>        443/TCP   104d

```

**Comment by mdditt2000** [comment]
SR 1-6599740401

Recreated here https://github.com/mdditt2000/kubernetes-1-18/tree/master/cis%202.1/github/1-6599740401

CIS does not like the service. 

```
Receiving the following error message

2020/08/18 04:15:12 [INFO] [CORE] Watching Ingress resources.
2020/08/18 04:15:12 [INFO] [CORE] Watching ConfigMap resources.
2020/08/18 04:15:12 [INFO] [CORE] Handling ConfigMap resource events.
2020/08/18 04:15:12 [INFO] [CORE] Handling Ingress resource events.
2020/08/18 04:15:12 [INFO] [CORE] Registered BigIP Metrics
2020/08/18 04:15:12 [INFO] Traceback (most recent call last):
2020/08/18 04:15:12 [INFO]   File "/app/src/f5-ctlr-agent/f5_ctlr_agent/bigipconfigdriver.py", line 325, in _do_reset
2020/08/18 04:15:12 [INFO]     incomplete = self._update_cccl(config)
2020/08/18 04:15:12 [INFO]   File "/app/src/f5-ctlr-agent/f5_ctlr_agent/bigipconfigdriver.py", line 397, in _update_cccl
2020/08/18 04:15:12 [INFO]     incomplete += mgr._apply_net_config(cfg_net)
2020/08/18 04:15:12 [INFO]   File "/app/src/f5-ctlr-agent/f5_ctlr_agent/bigipconfigdriver.py", line 121, in _apply_net_config
2020/08/18 04:15:12 [INFO]     return self._cccl.apply_net_config(config)
2020/08/18 04:15:12 [INFO]   File "/app/src/f5-cccl/f5_cccl/api.py", line 102, in apply_net_config
2020/08/18 04:15:12 [INFO]     return self._service_manager.apply_net_config(services)
2020/08/18 04:15:12 [INFO]   File "/app/src/f5-cccl/f5_cccl/service/manager.py", line 698, in apply_net_config
2020/08/18 04:15:12 [INFO]     default_route_domain = self._bigip.get_default_route_domain()
2020/08/18 04:15:12 [INFO]   File "/app/src/f5-cccl/f5_cccl/bigip.py", line 433, in get_default_route_domain
2020/08/18 04:15:12 [INFO]     name=self._partition)
2020/08/18 04:15:12 [INFO]   File "/usr/local/lib/python3.6/site-packages/f5/bigip/resource.py", line 1054, in load
2020/08/18 04:15:12 [INFO]     return self._load(**kwargs)
2020/08/18 04:15:12 [INFO]   File "/usr/local/lib/python3.6/site-packages/f5/bigip/resource.py", line 1033, in _load
2020/08/18 04:15:12 [INFO]     response = refresh_session.get(base_uri, **kwargs)
2020/08/18 04:15:12 [INFO]   File "/usr/local/lib/python3.6/site-packages/icontrol/session.py", line 271, in wrapper
2020/08/18 04:15:12 [INFO]     raise iControlUnexpectedHTTPError(error_message, response=response)
2020/08/18 04:15:12 [INFO] icontrol.exceptions.iControlUnexpectedHTTPError: 503 Unexpected Error: Service Unavailable for uri: https://192.168.200.92:443/mgmt/tm/auth/partition/Common
2020/08/18 04:15:12 [INFO] Text: '{"code":503,"message":"There is an active asynchronous task executing.","errorStack":[],"apiError":32964609}'
[root@k8s-1-18-master ~]#
```

Chat with PD

**Comment by mdditt2000** [comment]
Good morning @f5minions 

Issue is the kubernetes services has no endpoints and therefore CIS logs the following error message 

panic: runtime error: invalid memory address or nil pointer dereference [recovered]
panic: runtime error: invalid memory address or **nil pointer dereference**

Speaking with development CIS needs a valid endpoint for Ingress. Providing Ingress to the Kubernetes service is not something that CIS supports today. 

Please can you provide more information on why you want to create a Ingress to a service with no endpoints? I need to understand this use-case.

**Comment by mdditt2000** [comment]
Filed Jira in the CIS backlog for PM tracking. CONTCNTR-2085

**Comment by mdditt2000** [comment]
Crash is resolved and image will be provided in a few days. Health monitor is not a requirement at this time. Please confirm if that is ok.

**Comment by mdditt2000** [comment]
Merge request - https://github.com/F5Networks/k8s-bigip-ctlr/pull/1575

**Comment by mdditt2000** [comment]
Closed SR. This issue is complete and will be released this week in CIS 2.2.1
